code/game/machinery/doors/airlock_types.dm

```diff
@@ -631,16 +631,27 @@
 	Misc Airlocks
 */
 
-/obj/machinery/door/airlock/glass_large
+/obj/machinery/door/airlock/multi_tile/glass
+	multi_tile = TRUE
 	name = "large glass airlock"
 	icon = 'icons/obj/doors/airlocks/glass_large/glass_large.dmi'
 	overlays_file = 'icons/obj/doors/airlocks/glass_large/overlays.dmi'
 	opacity = FALSE
-	assemblytype = null
+	assemblytype = /obj/structure/door_assembly/multi_tile/glass
 	glass = TRUE
 	bound_width = 64 // 2x1
 
-/obj/machinery/door/airlock/glass_large/narsie_act()
+/obj/structure/fluff/airlock_filler
+	icon = null
+	icon_state = null
+	density = TRUE
+	opacity = TRUE
+	anchored = TRUE
+	can_atmos_pass = ATMOS_PASS_DENSITY
+	/// The door/airlock this fluff panel is attached to
+	var/obj/machinery/door/filled_airlock
```

Comment:
I know that the airlock qdels this when it is itself deleted, but I think you should register to the qdeleting signal from the airlock and clear this reference there just in case someone is doing something funny.